DEFEDIT-842 Handle tile-maps using out-of-range cells

ragnard-king opened the pull request:
- show error if tile source contains fewer tiles than the tile map uses
- render missing tiles as whole tile source (ie. 0-1 uvs)
- fix range bug when creating a selection brush from one or more out-of-range tiles 
- disable erase tool when no tile source has been set
- don't start paint/select operations when no tile source has been set

Fixes https://github.com/defold/editor2-issues/issues/603

ragnard-king committed:
DEFEDIT-842 Handle tile-maps using out-of-range cells

matgis-king reviewed (commented):
Reviewed. Potential off-by-one in error message, but looks good otherwise!

matgis-king reviewed (changes requested):
Re-reviewed. Sadly the new error message has bad grammar.

ragnard-king committed:
DEFEDIT-842 Review fixes

- fix incorrect grammar in error message
- rename active-tile-map? and active-scene-view? as they aren't predicates

matgis-king reviewed (approved):
Ship it!
